make native stopped working for mac64

`make native-all` fails with the following error:
 
```
cd target/snappy-1.1.7-Mac-x86_64 && cmake  ../../target/snappy-1.1.7
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.1 or higher is required.  You are running version 3.0.2


-- Configuring incomplete, errors occurred!
Makefile:83: recipe for target 'target/snappy-1.1.7-Mac-x86_64/CMakeCache.txt' failed
make: *** [target/snappy-1.1.7-Mac-x86_64/CMakeCache.txt] Error 1
make: *** [mac64] Error 2
```

The multiarch/crossbuild image hungs when running CMake... 

So I tried `make native` on Mac, but I hit this issue: 
```
touch target/snappy-1.1.7-Mac-x86_64/CMakeCache.txt
c++ -arch x86_64  -Ilib/inc_mac -I/Library/Java/JavaVirtualMachines/jdk-11.0.8.jdk/Contents/Home/include -O2 -fPIC -mmacosx-version-min=10.5 -fvisibility=hidden -stdlib=libstdc++ -DHAVE_CONFIG_H -I"/Library/Java/JavaVirtualMachines/jdk-11.0.8.jdk/Contents/Home/include/darwin" -Itarget/snappy-1.1.7 -Itarget/snappy-1.1.7-Mac-x86_64 -Itarget/bitshuffle-0.3.2/src -c target/snappy-1.1.7/snappy-sinksource.cc -o target/snappy-1.1.7-Mac-x86_64/snappy-sinksource.o
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++
      standard library instead [-Wstdlibcxx-not-found]
1 warning generated.
c++ -arch x86_64  -Ilib/inc_mac -I/Library/Java/JavaVirtualMachines/jdk-11.0.8.jdk/Contents/Home/include -O2 -fPIC -mmacosx-version-min=10.5 -fvisibility=hidden -stdlib=libstdc++ -DHAVE_CONFIG_H -I"/Library/Java/JavaVirtualMachines/jdk-11.0.8.jdk/Contents/Home/include/darwin" -Itarget/snappy-1.1.7 -Itarget/snappy-1.1.7-Mac-x86_64 -Itarget/bitshuffle-0.3.2/src -c target/snappy-1.1.7/snappy-stubs-internal.cc -o target/snappy-1.1.7-Mac-x86_64/snappy-stubs-internal.o
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++
      standard library instead [-Wstdlibcxx-not-found]
target/snappy-1.1.7/snappy-stubs-internal.cc:29:10: fatal error: 'algorithm' file not found
#include <algorithm>
         ^~~~~~~~~~~
1 warning and 1 error generated.
make: *** [target/snappy-1.1.7-Mac-x86_64/snappy-stubs-internal.o] Error 1
```

Related https://github.com/pandas-dev/pandas/issues/23424

--- a/Makefile
+++ b/Makefile
@@ -144,7 +144,7 @@ native: jni-header snappy-header $(NATIVE_DLL)
 native-nocmake: jni-header $(NATIVE_DLL)
 snappy: native $(TARGET)/$(snappy-jar-version).jar
 
-native-all: native win32 win64 native-arm linux32 linux64 linux-ppc64le
+native-all: native mac64 win32 win64 native-arm linux32 linux64 linux-ppc64le
 
 $(NATIVE_DLL): $(SNAPPY_OUT)/$(LIBNAME)
 	@mkdir -p $(@D)
@@ -172,7 +172,6 @@ win64: jni-header
 mac32: jni-header
 	$(MAKE) native OS_NAME=Mac OS_ARCH=x86
 
-# TODO: This command no longer works as the crossbuild docker image contains no cross compiler for Mac OS X (Darwin)
 mac64: jni-header
 	docker run -it $(DOCKER_RUN_OPTS) -v $$PWD:/workdir -e CROSS_TRIPLE=x86_64-apple-darwin xerial/crossbuild make clean-native native OS_NAME=Mac OS_ARCH=x86_64
 

--- a/docker/Makefile
+++ b/docker/Makefile
@@ -6,4 +6,4 @@ centos5-image:
 	# docker push xerial/centos5-linux-x86_64:latest
 
 multiarch-crossbuild-image:
-	docker build https://github.com/multiarch/crossbuild.git -t xerial/crossbuild:latest
+	docker build https://github.com/iwasakims/crossbuild.git#fix-osxcross-cmake -t xerial/crossbuild:latest

